test/unit/plugins/doca_memos/doca_memos_backend_test.cpp

```
@@ -0,0 +1,1809 @@
+
+    for (int req = 0; req < 3; req++) {
+        local_dlists.emplace_back(DRAM_SEG);
+        remote_dlists.emplace_back(OBJ_SEG);
+        buffers.emplace_back();
+        metadatas.emplace_back();
+
+        for (int i = 0; i < 2; i++) {
+            nixlMetaDesc local_desc, remote_desc;
+            void *buf = malloc(1024);
+            // Write a unique marker to verify ordering
+            memset(buf, 'A' + req, 1024);
```

Cppcheck (2.20.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)